ykcompile/src/lib.rs

```diff
@@ -214,7 +214,7 @@ impl<TT> TraceCompiler<TT> {
             // An array is copyable if its elements are.
             Ty::Array(ety) => Self::is_copyable(ety),
             Ty::Slice(ety) => Self::is_copyable(ety),
-            Ty::Ref(_) | Ty::Bool => true,
+            Ty::Ref(_) | Ty::Bool | Ty::Char => true,
```

Comment:
Are these unicode chars? If so they may not fit in a register, but I think we can address that later.

Reply:
They are unicode chars. Those are represented as 32bit integers, of which only 24bit are used. The rest must be all zeros.

Reply:
So not UTF-8?

I was expecting chars to be variable width utf-8 encoded.

Reply:
Good point.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c87c93dc6af84ebf7cbaaa1d1a04e8f9

```rust
fn main() {
    assert_eq!(std::mem::size_of::<char>(), 4);
}
```

Reply:
Ah, yes, `chars` are 4 bytes, which is addressed in `test_cast`.